pallets/clients-info/src/lib.rs

```diff
@@ -72,6 +73,26 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Pallet<T> {
+		fn check_non_root_rights(origin: OriginFor<T>) -> Result<T::AccountId, DispatchError> {
+			let origin_account_id = ensure_signed(origin)?;
+			ensure!(
+				<AuthorizedAccounts<T>>::contains_key(&origin_account_id),
+				Error::<T>::ThisAccountIdIsNotAuthorized
+			);
+
+			Ok(origin_account_id)
+		}
+
+		fn check_origin_rights(origin: OriginFor<T>) -> DispatchResult {
+			if ensure_root(origin.clone()).is_err() {
+				let _ = Pallet::<T>::check_non_root_rights(origin)?;
+			};
+
+			Ok(())
+		}
+	}
```

Comment:
Can we move this code below the code block for extrinsics? It does not matter really, but I would expect to first read the 'important' functions ie. extrinsics, before reading some helper functions. Or would you prefer to keep it like this @b-yap?